modules/apps/foundation/captcha/captcha-api/src/main/java/com/liferay/captcha/verify/CaptchaPropertiesVerifyProcess.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.captcha.verify;
+
+import com.liferay.captcha.configuration.CaptchaConfiguration;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.HashMapDictionary;
+import com.liferay.portal.kernel.util.LoggingTimer;
+import com.liferay.portal.kernel.util.PrefsProps;
+import com.liferay.portal.kernel.util.StringPool;
+import com.liferay.portal.verify.VerifyProcess;
+
+import java.util.Dictionary;
+
+import javax.portlet.PortletPreferences;
+
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(
+	immediate = true,
+	property = {"verify.process.name=com.liferay.captcha.verify"},
+	service = VerifyProcess.class
+)
+public class CaptchaPropertiesVerifyProcess extends VerifyProcess {
+
+	@Override
+	protected void doVerify() throws Exception {
+		upgradeConfiguration();
+	}
+
+	@Reference(unbind = "-")
+	protected void setPrefsProps(PrefsProps prefsProps) {
+		_prefsProps = prefsProps;
+	}
+
+	protected void upgradeConfiguration() throws Exception {
+		try (LoggingTimer loggingTimer = new LoggingTimer()) {
+			Dictionary<String, Object> dictionary = new HashMapDictionary<>();
```

Comment:
This variable should be named `properties` to stay consistent with its purpose.  For example, this dictionary is retrieved from the configuration by calling `configuration.getProperties()`.